apps/backend/src/modules/contributors/contributors.service.ts

```diff
@@ -0,0 +1,78 @@
+import { ExceptionMessage } from "~/libs/enums/enums.js";
+import { HTTPCode } from "~/libs/modules/http/http.js";
+import { type Service } from "~/libs/types/types.js";
+
+import { ContributorEntity } from "./contributor.entity.js";
+import { type ContributorRepository } from "./contributor.repository.js";
+import { ContributorError } from "./libs/exceptions/exceptions.js";
+import {
+	type ContributorCreateRequestDto,
+	type ContributorGetAllItemResponseDto,
+} from "./libs/types/types.js";
+
+class ContributorService implements Service {
+	private contributorRepository: ContributorRepository;
+
+	public constructor(contributorRepository: ContributorRepository) {
+		this.contributorRepository = contributorRepository;
+	}
+
+	public async create(
+		payload: ContributorCreateRequestDto,
+	): Promise<ContributorGetAllItemResponseDto> {
+		const { name } = payload;
+
+		const existingContributor =
+			await this.contributorRepository.findByName(name);
+
+		if (existingContributor) {
+			throw new ContributorError({
+				message: ExceptionMessage.CONTRIBUTOR_NAME_USED,
+				status: HTTPCode.CONFLICT,
+			});
+		}
```

Comment:
Contributor name can be non unique, we will identify different contributors by different git emails
```suggestion
```